doc/releases/changelog-dev.md

```diff
@@ -59,7 +61,7 @@
 
 <h3>Deprecations 👋</h3>
 
-* Matrix and tensor products between `PauliWord` and `PauliSentence` instances are done using the `@` operator, `*` will be used only for scalar multiplication.
+* Matrix and tensor products between `PauliWord` and `PauliSentence` instances are done using the `@` operator, `*` will be used only for scalar multiplication. Note also the breaking change that the product of two `PauliWord` instances now returns a `PauliSentence` instead of a tuple `(new_word, coeff)`.
```

Comment:
This is making me wonder whether we need a deprecation :thinking:
Could it be that `@` gives a PauliSentence and `*` gives the old behaviour?

Reply:
Hmm, I'd find it confusing to have two operators do the same thing (matrix/tensor product) but with two different return types 🤔

Reply:
We can discuss more in our kickoff tomorrow, I'm just trying to find the best way to avoid sudden breaking changes. Though admittedly I'm not sure how many users are currently taking products of `PauliWord`s.

Reply:
> Could it be that `@` gives a PauliSentence and `*` gives the old behaviour?

Also, this suggestion was meant just for the lifetime of the deprecation, we'd also remove that behaviour in a future release.

Reply:
Ahh okay! Yes that makes sense, this is already the case with this change in this PR https://github.com/PennyLaneAI/pennylane/pull/5054/files#diff-a289f572837e011b4483e56087180b99f765b8dadc9505e3b1ed8be55481a9f7R257

Reply:
updated the warning, might be a bit lengthy though 😬

Reply:
I think it seems good!